homeassistant/components/media_player/openhome.py

```diff
@@ -0,0 +1,221 @@
+DEVICES = []
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup Openhome Platform."""
+    from openhomedevice.Device import Device
+
+    if discovery_info:
+        _LOGGER.info('Openhome device found, (%s)', discovery_info[0])
+        device = Device(discovery_info[1])
+
+        # if device has already been discovered
+        if device.Uuid() in [x.unique_id for x in DEVICES]:
+            return True
+
+        device = OpenhomeDevice(hass, device)
+
+        add_devices([device], True)
+        DEVICES.append(device)
+
+        return True
+
+    return True
+
+
+class OpenhomeDevice(MediaPlayerDevice):
+    """Representation of an Openhome device."""
+
+    def __init__(self, hass, device):
+        """Initialise the Openhome device."""
+        self.hass = hass
+        self._device = device
+        self._track_information = {}
+        self._in_standby = None
+        self._transport_state = None
+        self._track_information = None
+        self._volume_level = None
+        self._volume_muted = None
+        self._supported_features = SUPPORT_OPENHOME
+        self._source_names = list()
+        self._name = None
+        self._state = STATE_PLAYING
+
+    def update(self):
+        """Update state of device."""
+        self._in_standby = self._device.IsInStandby()
+        self._transport_state = self._device.TransportState()
+        self._track_information = self._device.TrackInfo()
+        self._volume_level = self._device.VolumeLevel()
+        self._volume_muted = self._device.IsMuted()
+        self._source = self._device.Source()
+        self._source_index = {}
+        self._name = self._device.Room().decode('utf-8')
+        self._supported_features = SUPPORT_OPENHOME
+
+        for source in self._device.Sources():
+            self._source_names.append(source["name"])
+            self._source_index[source["name"]] = source["index"]
+
+        if self._source["type"] == "Radio":
+            if self._transport_state in ('Playing', 'Buffering'):
+                self._supported_features |= SUPPORT_STOP
+            else:
+                self._supported_features |= SUPPORT_PLAY
+        if self._source["type"] == "Playlist":
+            self._supported_features |= SUPPORT_PREVIOUS_TRACK |\
+                SUPPORT_NEXT_TRACK
+            if self._transport_state in ('Playing', 'Buffering'):
+                self._supported_features |= SUPPORT_PAUSE
+            else:
+                self._supported_features |= SUPPORT_PLAY
+
+        if self._in_standby:
+            self._state = STATE_OFF
+        elif self._transport_state == 'Paused':
+            self._state = STATE_PAUSED
+        elif self._transport_state in ('Playing', 'Buffering'):
+            self._state = STATE_PLAYING
+        elif self._transport_state == 'Stopped':
+            self._state = STATE_IDLE
+        else:
+            self._state = STATE_PLAYING
```

Comment:
Based on your other comment, it would be nice just to add a comment above this line, explaining why else returns Playing. Hopefully then the next person isn't confused. :)